## Combining Objects
## join
## merge
## concat

In [1]:
import pandas as pd

In [25]:
names = pd.read_csv('names.csv')
names

Name  Age
0  Cornelia   70
1     Abbas   69
2  Penelope    4
3      Niko    2

In [26]:
ls = ['Abdulla', 24]

names.loc[6] = ls
names

Name  Age
0  Cornelia   70
1     Abbas   69
2  Penelope    4
3      Niko    2
6   Abdulla   24

In [6]:
names.loc['five'] = ['Josh', 42]

names

Name  Age
0     Cornelia   70
1        Abbas   69
2     Penelope    4
3         Niko    2
6      Abdulla   24
five      Josh   42

In [7]:
len(names)

6

In [8]:
names.loc[len(names)] = pd.Series({'Name': 'Alex', 'Age': 44})

names

Name  Age
0     Cornelia   70
1        Abbas   69
2     Penelope    4
3         Niko    2
6         Alex   44
five      Josh   42

# concat

In [27]:
stocks_2016 = pd.read_csv('stocks_2016.csv')
stocks_2017 = pd.read_csv('stocks_2017.csv')

In [28]:
stocks_2016

Symbol  Shares  Low  High
0   AAPL      80   95   110
1   TSLA      50   80   130
2    WMT      40   55    70

In [29]:
stocks_2017

Symbol  Shares  Low  High
0   AAPL      50  120   140
1     GE     100   30    40
2    IBM      87   75    95
3    SLB      20   55    85
4    TXN     500   15    23
5   TSLA     100  100   300

In [31]:
ls = [stocks_2016, stocks_2017]

pd.concat(ls)

Symbol  Shares  Low  High
0   AAPL      80   95   110
1   TSLA      50   80   130
2    WMT      40   55    70
0   AAPL      50  120   140
1     GE     100   30    40
2    IBM      87   75    95
3    SLB      20   55    85
4    TXN     500   15    23
5   TSLA     100  100   300

In [32]:
pd.concat(ls, keys=['2016', '2017'], names=['Year', 'Symbol'])

Symbol  Shares  Low  High
Year Symbol                          
2016 0        AAPL      80   95   110
     1        TSLA      50   80   130
     2         WMT      40   55    70
2017 0        AAPL      50  120   140
     1          GE     100   30    40
     2         IBM      87   75    95
     3         SLB      20   55    85
     4         TXN     500   15    23
     5        TSLA     100  100   300

In [33]:
pd.concat(ls, keys=['2016', '2017'], axis='columns', names=['Year', None])

Year   2016                       2017                 
     Symbol Shares   Low   High Symbol Shares  Low High
0      AAPL   80.0  95.0  110.0   AAPL     50  120  140
1      TSLA   50.0  80.0  130.0     GE    100   30   40
2       WMT   40.0  55.0   70.0    IBM     87   75   95
3       NaN    NaN   NaN    NaN    SLB     20   55   85
4       NaN    NaN   NaN    NaN    TXN    500   15   23
5       NaN    NaN   NaN    NaN   TSLA    100  100  300

In [39]:
pd.concat(ls, join='inner', keys=['2016', '2017'], axis=1, names=['Year', None])

Year   2016                   2017                 
     Symbol Shares Low High Symbol Shares  Low High
0      AAPL     80  95  110   AAPL     50  120  140
1      TSLA     50  80  130     GE    100   30   40
2       WMT     40  55   70    IBM     87   75   95

# Join and Merge

In [41]:
years = 2016, 2017, 2018

stocks_years = [pd.read_csv(f'stocks_{year}.csv', index_col='Symbol') for year in years]

In [42]:
stocks_2016, stocks_2017, stocks_2018 = stocks_years

In [43]:
stocks_2018

Shares  Low  High
Symbol                   
AAPL        40  135   170
AMZN         8  900  1125
TSLA        50  220   400

In [45]:
stocks_2017

Shares  Low  High
Symbol                   
AAPL        50  120   140
GE         100   30    40
IBM         87   75    95
SLB         20   55    85
TXN        500   15    23
TSLA       100  100   300

In [48]:
stocks_2016.join(stocks_2017, on= 'Symbol', lsuffix='_2016', rsuffix='_2017', how= 'inner')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017
Symbol                                                                    
AAPL             80        95        110           50       120        140
TSLA             50        80        130          100       100        300

In [53]:
stocks_2016.join(stocks_2017, lsuffix='_2016', rsuffix='_2017', how= 'left')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017
Symbol                                                                    
AAPL             80        95        110         50.0     120.0      140.0
TSLA             50        80        130        100.0     100.0      300.0
WMT              40        55         70          NaN       NaN        NaN

In [79]:
stocks_2016.join(stocks_2017, lsuffix='_2016', rsuffix='_2017', how= 'cross')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017
0            80        95        110           50       120        140
1            80        95        110          100        30         40
2            80        95        110           87        75         95
3            80        95        110           20        55         85
4            80        95        110          500        15         23
5            80        95        110          100       100        300
6            50        80        130           50       120        140
7            50        80        130          100        30         40
8            50        80        130           87        75         95
9            50        80        130           20        55         85
10           50        80        130          500        15         23
11           50        80        130          100       100        300
12           40        55         70           50       120        140
13           40        55         70          100        30         40
14           40        55         70           87        75         95
15           40        55         70           20        55         85
16           40        55         70          500        15         23
17           40        55         70          100       100        300

In [56]:
stocks_2016.join(stocks_2017, lsuffix='_2016', rsuffix='_2017', how= 'outer')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017
Symbol                                                                    
AAPL           80.0      95.0      110.0         50.0     120.0      140.0
GE              NaN       NaN        NaN        100.0      30.0       40.0
IBM             NaN       NaN        NaN         87.0      75.0       95.0
SLB             NaN       NaN        NaN         20.0      55.0       85.0
TSLA           50.0      80.0      130.0        100.0     100.0      300.0
TXN             NaN       NaN        NaN        500.0      15.0       23.0
WMT            40.0      55.0       70.0          NaN       NaN        NaN

In [63]:
other = [
    stocks_2017.add_suffix('_2017'),
    stocks_2018.add_suffix('_2018')
]

#stocks_2016.add_suffix('_2016').join(stocks_2017.add_suffix(('_2017')), how='outer').join(stocks_2018.add_suffix('_2018'))
stocks_2016.add_suffix('_2016').join(other, how= 'outer')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017  \
Symbol                                                                       
AAPL           80.0      95.0      110.0         50.0     120.0      140.0   
TSLA           50.0      80.0      130.0        100.0     100.0      300.0   
WMT            40.0      55.0       70.0          NaN       NaN        NaN   
GE              NaN       NaN        NaN        100.0      30.0       40.0   
IBM             NaN       NaN        NaN         87.0      75.0       95.0   
SLB             NaN       NaN        NaN         20.0      55.0       85.0   
TXN             NaN       NaN        NaN        500.0      15.0       23.0   
AMZN            NaN       NaN        NaN          NaN       NaN        NaN   

        Shares_2018  Low_2018  High_2018  
Symbol                                    
AAPL           40.0     135.0      170.0  
TSLA           50.0     220.0      400.0  
WMT             NaN       NaN        NaN  
GE              NaN       NaN        NaN  
IBM             NaN       NaN        NaN  
SLB             NaN       NaN        NaN  
TXN             NaN       NaN        NaN  
AMZN            8.0     900.0     1125.0

In [71]:
stocks_2016.merge(stocks_2017, left_index=True, right_index=True, how='inner')

Shares_x  Low_x  High_x  Shares_y  Low_y  High_y
Symbol                                                  
AAPL          80     95     110        50    120     140
TSLA          50     80     130       100    100     300

In [78]:
df1 = pd.DataFrame({'Value_A': [10, 20, 30]}, index=['a', 'b', 'c'])
df2 = pd.DataFrame({'Value_B': [100, 200, 300]}, index=['b', 'c', 'd'])

merged = df1.merge(df2, left_index=True, right_index=True)
merged

Value_A  Value_B
b       20      100
c       30      200

In [73]:
df2

Value_B
b      100
c      200
d      300

In [75]:
df1

Value_A
a       10
b       20
c       30

In [84]:
stocks_2016.merge(stocks_2017, 
                  left_index=True, 
                  right_index=True, 
                  how='outer', 
                  suffixes=('_2016', '_2017')
                  ).merge(stocks_2018.add_suffix('_2018'), 
                          left_index=True, 
                          right_index=True, 
                          how='outer')

Shares_2016  Low_2016  High_2016  Shares_2017  Low_2017  High_2017  \
Symbol                                                                       
AAPL           80.0      95.0      110.0         50.0     120.0      140.0   
AMZN            NaN       NaN        NaN          NaN       NaN        NaN   
GE              NaN       NaN        NaN        100.0      30.0       40.0   
IBM             NaN       NaN        NaN         87.0      75.0       95.0   
SLB             NaN       NaN        NaN         20.0      55.0       85.0   
TSLA           50.0      80.0      130.0        100.0     100.0      300.0   
TXN             NaN       NaN        NaN        500.0      15.0       23.0   
WMT            40.0      55.0       70.0          NaN       NaN        NaN   

        Shares_2018  Low_2018  High_2018  
Symbol                                    
AAPL           40.0     135.0      170.0  
AMZN            8.0     900.0     1125.0  
GE              NaN       NaN        NaN  
IBM             NaN       NaN        NaN  
SLB             NaN       NaN        NaN  
TSLA           50.0     220.0      400.0  
TXN             NaN       NaN        NaN  
WMT             NaN       NaN        NaN

In [85]:
food_prices = pd.read_csv('food_prices.csv')
food_transactions = pd.read_csv('food_transactions.csv')

In [86]:
food_prices

item store  price  Date
0    pear     A   0.99  2017
1    pear     B   1.99  2017
2   peach     A   2.99  2017
3   peach     B   3.49  2017
4  banana     A   0.39  2017
5  banana     B   0.49  2017
6   steak     A   5.99  2017
7   steak     B   6.99  2017
8   steak     B   4.99  2015

In [87]:
food_transactions

custid     item store  quantity
0       1     pear     A         5
1       1   banana     A        10
2       2    steak     B         3
3       2     pear     B         1
4       2    peach     B         2
5       2    steak     B         1
6       2  coconut     B         4

In [89]:
food_transactions.merge(food_prices, on=['item', 'store'])

custid    item store  quantity  price  Date
0       1    pear     A         5   0.99  2017
1       1  banana     A        10   0.39  2017
2       2   steak     B         3   6.99  2017
3       2   steak     B         3   4.99  2015
4       2    pear     B         1   1.99  2017
5       2   peach     B         2   3.49  2017
6       2   steak     B         1   6.99  2017
7       2   steak     B         1   4.99  2015

In [90]:
food_value = food_prices.query('Date==2017').set_index(['item', 'store'])

In [96]:
food_transactions.set_index(['item', 'store'])

custid  quantity
item    store                  
pear    A           1         5
banana  A           1        10
steak   B           2         3
pear    B           2         1
peach   B           2         2
steak   B           2         1
coconut B           2         4

In [99]:
food_value

price  Date
item   store             
pear   A       0.99  2017
       B       1.99  2017
peach  A       2.99  2017
       B       3.49  2017
banana A       0.39  2017
       B       0.49  2017
steak  A       5.99  2017
       B       6.99  2017

In [106]:
food_transactions.join(food_value, on=['item', 'store'], how='inner')

custid    item store  quantity  price  Date
0       1    pear     A         5   0.99  2017
1       1  banana     A        10   0.39  2017
2       2   steak     B         3   6.99  2017
3       2    pear     B         1   1.99  2017
4       2   peach     B         2   3.49  2017
5       2   steak     B         1   6.99  2017

In [114]:
from sqlalchemy import create_engine

engine = create_engine('sqlite:///data/chinook.db')

In [115]:
tracks = pd.read_sql_table('tracks', engine)

tracks

TrackId                                               Name  AlbumId  \
0           1            For Those About To Rock (We Salute You)        1   
1           2                                  Balls to the Wall        2   
2           3                                    Fast As a Shark        3   
3           4                                  Restless and Wild        3   
4           5                               Princess of the Dawn        3   
...       ...                                                ...      ...   
3498     3499  Pini Di Roma (Pinien Von Rom) \ I Pini Della V...      343   
3499     3500  String Quartet No. 12 in C Minor, D. 703 "Quar...      344   
3500     3501               L'orfeo, Act 3, Sinfonia (Orchestra)      345   
3501     3502  Quintet for Horn, Violin, 2 Violas, and Cello ...      346   
3502     3503                                      Koyaanisqatsi      347   

      MediaTypeId  GenreId                                           Composer  \
0               1        1          Angus Young, Malcolm Young, Brian Johnson   
1               2        1                                               None   
2               2        1  F. Baltes, S. Kaufman, U. Dirkscneider & W. Ho...   
3               2        1  F. Baltes, R.A. Smith-Diesel, S. Kaufman, U. D...   
4               2        1                         Deaffy & R.A. Smith-Diesel   
...           ...      ...                                                ...   
3498            2       24                                               None   
3499            2       24                                     Franz Schubert   
3500            2       24                                 Claudio Monteverdi   
3501            2       24                            Wolfgang Amadeus Mozart   
3502            2       10                                       Philip Glass   

      Milliseconds     Bytes  UnitPrice  
0           343719  11170334       0.99  
1           342562   5510424       0.99  
2           230619   3990994       0.99  
3           252051   4331779       0.99  
4           375418   6290521       0.99  
...            ...       ...        ...  
3498        286741   4718950       0.99  
3499        139200   2283131       0.99  
3500         66639   1189062       0.99  
3501        221331   3665114       0.99  
3502        206005   3305164       0.99  

[3503 rows x 9 columns]

In [116]:
genres = pd.read_sql_table('genres', engine)
genres

GenreId                Name
0         1                Rock
1         2                Jazz
2         3               Metal
3         4  Alternative & Punk
4         5       Rock And Roll
5         6               Blues
6         7               Latin
7         8              Reggae
8         9                 Pop
9        10          Soundtrack
10       11          Bossa Nova
11       12      Easy Listening
12       13         Heavy Metal
13       14            R&B/Soul
14       15   Electronica/Dance
15       16               World
16       17         Hip Hop/Rap
17       18     Science Fiction
18       19            TV Shows
19       20    Sci Fi & Fantasy
20       21               Drama
21       22              Comedy
22       23         Alternative
23       24           Classical
24       25               Opera

In [130]:
genres.merge(tracks[['Milliseconds', 'GenreId']], on='GenreId', how='right').drop('GenreId', axis='columns')

Name  Milliseconds
0           Rock        343719
1           Rock        342562
2           Rock        230619
3           Rock        252051
4           Rock        375418
...          ...           ...
3498   Classical        286741
3499   Classical        139200
3500   Classical         66639
3501   Classical        221331
3502  Soundtrack        206005

[3503 rows x 2 columns]

In [131]:
cust = pd.read_sql_table('customers', engine, columns=['CustomerId', 'FirstName', 'LastName'])
invoice = pd.read_sql_table('invoices', engine, columns=['InvoiceId', 'CustomerId'])
invoice_items = pd.read_sql_table('invoice_items', engine, columns=['InvoiceId', 'UnitPrice', 'Quantity'])

In [135]:
cust.merge(invoice, on= 'CustomerId', how='inner').merge(invoice_items, on='InvoiceId')

CustomerId FirstName    LastName  InvoiceId  UnitPrice  Quantity
0              1      Luís   Gonçalves         98       1.99         1
1              1      Luís   Gonçalves         98       1.99         1
2              1      Luís   Gonçalves        121       0.99         1
3              1      Luís   Gonçalves        121       0.99         1
4              1      Luís   Gonçalves        121       0.99         1
...          ...       ...         ...        ...        ...       ...
2235          59      Puja  Srivastava        284       0.99         1
2236          59      Puja  Srivastava        284       0.99         1
2237          59      Puja  Srivastava        284       0.99         1
2238          59      Puja  Srivastava        284       0.99         1
2239          59      Puja  Srivastava        284       0.99         1

[2240 rows x 6 columns]

In [136]:
sql_string = '''
    select * from invoices where CustomerId = 3
'''

pd.read_sql_query(sql_string, engine)

InvoiceId  CustomerId          InvoiceDate     BillingAddress BillingCity  \
0         99           3  2010-03-11 00:00:00  1498 rue Bélanger    Montréal   
1        110           3  2010-04-21 00:00:00  1498 rue Bélanger    Montréal   
2        165           3  2010-12-20 00:00:00  1498 rue Bélanger    Montréal   
3        294           3  2012-07-26 00:00:00  1498 rue Bélanger    Montréal   
4        317           3  2012-10-28 00:00:00  1498 rue Bélanger    Montréal   
5        339           3  2013-01-30 00:00:00  1498 rue Bélanger    Montréal   
6        391           3  2013-09-20 00:00:00  1498 rue Bélanger    Montréal   

  BillingState BillingCountry BillingPostalCode  Total  
0           QC         Canada           H2G 1A7   3.98  
1           QC         Canada           H2G 1A7  13.86  
2           QC         Canada           H2G 1A7   8.91  
3           QC         Canada           H2G 1A7   1.98  
4           QC         Canada           H2G 1A7   3.96  
5           QC         Canada           H2G 1A7   5.94  
6           QC         Canada           H2G 1A7   0.99

In [137]:
sql_string1 = '''
SELECT Name,
    time(avg(Milliseconds) / 1000, 'unixepoch') as avg_time
FROM (
    SELECT
        g.Name,
        t.Milliseconds
    FROM genres as g
    JOIN tracks as t on g.genreid == t.genreid
)
GROUP BY Name
ORDER BY avg_time'''

pd.read_sql_query(sql_string1, engine)

Name  avg_time
0        Rock And Roll  00:02:14
1                Opera  00:02:54
2          Hip Hop/Rap  00:02:58
3       Easy Listening  00:03:09
4           Bossa Nova  00:03:39
5             R&B/Soul  00:03:40
6                World  00:03:44
7                  Pop  00:03:49
8                Latin  00:03:52
9   Alternative & Punk  00:03:54
10          Soundtrack  00:04:04
11              Reggae  00:04:07
12         Alternative  00:04:24
13               Blues  00:04:30
14                Rock  00:04:43
15                Jazz  00:04:51
16           Classical  00:04:53
17         Heavy Metal  00:04:57
18   Electronica/Dance  00:05:02
19               Metal  00:05:09
20              Comedy  00:26:25
21            TV Shows  00:35:45
22               Drama  00:42:55
23     Science Fiction  00:43:45
24    Sci Fi & Fantasy  00:48:31

In [145]:
customers = pd.read_sql_query("SELECT CustomerId, FirstName, LastName FROM customers", engine)
invoices = pd.read_sql_query("SELECT InvoiceId, CustomerId, Total FROM invoices", engine)
albums = pd.read_sql_query("SELECT AlbumId, Title, ArtistId FROM albums", engine)
artists = pd.read_sql_query("SELECT ArtistId, Name FROM artists", engine)

customers.set_index("CustomerId")
invoices.set_index("CustomerId")


customers.join(invoices, how= 'inner', lsuffix='__', rsuffix='___')

CustomerId__  FirstName      LastName  InvoiceId  CustomerId___  Total
0              1       Luís     Gonçalves          1              2   1.98
1              2     Leonie        Köhler          2              4   3.96
2              3   François      Tremblay          3              8   5.94
3              4      Bjørn        Hansen          4             14   8.91
4              5  František   Wichterlová          5             23  13.86
5              6     Helena          Holý          6             37   0.99
6              7     Astrid        Gruber          7             38   1.98
7              8       Daan       Peeters          8             40   1.98
8              9       Kara       Nielsen          9             42   3.96
9             10    Eduardo       Martins         10             46   5.94
10            11  Alexandre         Rocha         11             52   8.91
11            12    Roberto       Almeida         12              2  13.86
12            13   Fernanda         Ramos         13             16   0.99
13            14       Mark       Philips         14             17   1.98
14            15   Jennifer      Peterson         15             19   1.98
15            16      Frank        Harris         16             21   3.96
16            17       Jack         Smith         17             25   5.94
17            18   Michelle        Brooks         18             31   8.91
18            19        Tim         Goyer         19             40  13.86
19            20        Dan        Miller         20             54   0.99
20            21      Kathy         Chase         21             55   1.98
21            22    Heather       Leacock         22             57   1.98
22            23       John        Gordon         23             59   3.96
23            24      Frank       Ralston         24              4   5.94
24            25     Victor       Stevens         25             10   8.91
25            26    Richard    Cunningham         26             19  13.86
26            27    Patrick          Gray         27             33   0.99
27            28      Julia       Barnett         28             34   1.98
28            29     Robert         Brown         29             36   1.98
29            30     Edward       Francis         30             38   3.96
30            31     Martha          Silk         31             42   5.94
31            32      Aaron      Mitchell         32             48   8.91
32            33      Ellie      Sullivan         33             57  13.86
33            34       João     Fernandes         34             12   0.99
34            35   Madalena       Sampaio         35             13   1.98
35            36     Hannah     Schneider         36             15   1.98
36            37       Fynn    Zimmermann         37             17   3.96
37            38     Niklas      Schröder         38             21   5.94
38            39    Camille       Bernard         39             27   8.91
39            40  Dominique      Lefebvre         40             36  13.86
40            41       Marc        Dubois         41             50   0.99
41            42      Wyatt        Girard         42             51   1.98
42            43   Isabelle       Mercier         43             53   1.98
43            44      Terhi    Hämäläinen         44             55   3.96
44            45   Ladislav        Kovács         45             59   5.94
45            46       Hugh      O'Reilly         46              6   8.91
46            47      Lucas       Mancini         47             15  13.86
47            48   Johannes  Van der Berg         48             29   0.99
48            49  Stanisław        Wójcik         49             30   1.98
49            50    Enrique         Muñoz         50             32   1.98
50            51     Joakim     Johansson         51             34   3.96
51            52       Emma         Jones         52             38   5.94
52            53       Phil  

In [142]:
customers.merge(invoices, how= 'inner')

CustomerId FirstName    LastName  InvoiceId  Total
0             1      Luís   Gonçalves         98   3.98
1             1      Luís   Gonçalves        121   3.96
2             1      Luís   Gonçalves        143   5.94
3             1      Luís   Gonçalves        195   0.99
4             1      Luís   Gonçalves        316   1.98
..          ...       ...         ...        ...    ...
407          59      Puja  Srivastava         45   5.94
408          59      Puja  Srivastava         97   1.99
409          59      Puja  Srivastava        218   1.98
410          59      Puja  Srivastava        229  13.86
411          59      Puja  Srivastava        284   8.91

[412 rows x 5 columns]

In [143]:
customers.set_index("CustomerId")
invoices.set_index("CustomerId")

InvoiceId  Total
CustomerId                  
2                   1   1.98
4                   2   3.96
8                   3   5.94
14                  4   8.91
23                  5  13.86
...               ...    ...
25                408   3.96
29                409   5.94
35                410   8.91
44                411  13.86
58                412   1.99

[412 rows x 2 columns]

In [146]:
artists

ArtistId                                               Name
0           1                                              AC/DC
1           2                                             Accept
2           3                                          Aerosmith
3           4                                  Alanis Morissette
4           5                                    Alice In Chains
..        ...                                                ...
270       271   Mela Tenenbaum, Pro Musica Prague & Richard Kapp
271       272                             Emerson String Quartet
272       273  C. Monteverdi, Nigel Rogers - Chiaroscuro; Lon...
273       274                                      Nash Ensemble
274       275                              Philip Glass Ensemble

[275 rows x 2 columns]

ArtistId                                                                  
25                               Milton Nascimento & Bebeto      NaN   NaN
26                                                  Azymuth      NaN   NaN
28                                            João Gilberto      NaN   NaN
29                                           Bebel Gilberto      NaN   NaN
30                                            Jorge Vercilo      NaN   NaN
...                                                     ...      ...   ...
192                      DJ Dolores & Orchestra Santa Massa      NaN   NaN
193                                               Seu Jorge      NaN   NaN
194                                    Sabotage E Instituto      NaN   NaN
195                                         Stereo Maracana      NaN   NaN
239       Academy of St. Martin in the Fields, Sir Nevil...      NaN   NaN

[71 rows x 3 columns]

In [147]:
albums

AlbumId                                              Title  ArtistId
0          1              For Those About To Rock We Salute You         1
1          2                                  Balls to the Wall         2
2          3                                  Restless and Wild         2
3          4                                  Let There Be Rock         1
4          5                                           Big Ones         3
..       ...                                                ...       ...
342      343                             Respighi:Pines of Rome       226
343      344  Schubert: The Late String Quartets & String Qu...       272
344      345                                Monteverdi: L'Orfeo       273
345      346                              Mozart: Chamber Music       274
346      347  Koyaanisqatsi (Soundtrack from the Motion Pict...       275

[347 rows x 3 columns]